test/cppsuite/test_harness/workload/database_operation.cpp

```diff
@@ -246,6 +246,75 @@ database_operation::read_operation(thread_context *tc)
         tc->transaction.rollback();
 }
 
+void
+database_operation::remove_operation(thread_context *tc)
+{
+    logger::log_msg(
+      LOG_INFO, type_string(tc->type) + " thread {" + std::to_string(tc->id) + "} commencing.");
+
+    /*
+     * We need two types of cursors. One cursor is a random cursor to randomly select a key and the
+     * other one is a standard cursor to remove the random key. This is required as the random
+     * cursor does not support the remove operation.
+     */
+    std::map<uint64_t, scoped_cursor> rnd_cursors, cursors;
+
+    /* Loop while the test is running. */
+    while (tc->running()) {
+        /*
+         * Sleep the period defined by the op_rate in the configuration. Do this at the start of the
+         * loop as it could be skipped by a subsequent continue call.
+         */
+        tc->sleep();
+
+        /* Choose a random collection to update. */
+        collection &coll = tc->db.get_random_collection();
+
+        /* Look for existing cursors in our cursor cache. */
+        if (cursors.find(coll.id) == cursors.end()) {
+            logger::log_msg(LOG_TRACE,
+              "Thread {" + std::to_string(tc->id) +
+                "} Creating cursor for collection: " + coll.name);
+            /* Open the two cursors for the chosen collection. */
+            scoped_cursor rnd_cursor =
+              tc->session.open_scoped_cursor(coll.name, "next_random=true");
+            rnd_cursors.emplace(coll.id, std::move(rnd_cursor));
+            scoped_cursor cursor = tc->session.open_scoped_cursor(coll.name);
+            cursors.emplace(coll.id, std::move(cursor));
+        }
+
+        /* Start a transaction if possible. */
+        tc->transaction.try_begin();
+
+        /* Get the cursor associated with the collection. */
+        scoped_cursor &rnd_cursor = rnd_cursors[coll.id];
+        scoped_cursor &cursor = cursors[coll.id];
+
+        /* Choose a random key to delete. */
+        const char *key_str;
+        int ret = rnd_cursor->next(rnd_cursor.get());
+        /* It is possible not to find anything if the collection is empty. */
+        testutil_assert(ret == 0 || ret == WT_NOTFOUND);
+        if (ret == WT_NOTFOUND)
+            continue;
+        testutil_check(rnd_cursor->get_key(rnd_cursor.get(), &key_str));
+        if (!tc->remove(cursor, coll.id, key_str)) {
```

Comment:
Can we just re-use the rnd_cursor here to perform remove?

Reply:
We cannot as this operation is not supported by the cursor, I added a comment [here](https://github.com/wiredtiger/wiredtiger/pull/7822/files#diff-d35d5f5a8892ee5cd7efdd7079e296da18b9223d0b563b5f92cc8d65f8fa8acaR255-R259).